tap_chargebee/streams/base.py

```diff
@@ -175,23 +175,24 @@ def sync_data(self):
             
             to_write = self.get_stream_data(records)
             
-            if self.ENTITY == 'event':
+            if self.config.get('include_deleted') not in ['false','False', False] and self.ENTITY == 'event':
                 for event in to_write:
                     if event["event_type"] == 'plan_deleted':
                         Util.plans.append(event['content']['plan'])
                     elif event['event_type'] == 'addon_deleted':
                         Util.addons.append(event['content']['addon'])
                     elif event['event_type'] == 'coupon_deleted':
                         Util.coupons.append(event['content']['coupon'])
-            if self.ENTITY == 'plan':
-                for plan in Util.plans:
-                    to_write.append(plan)
-            if self.ENTITY == 'addon':
-                for addon in Util.addons:
-                    to_write.append(addon)
-            if self.ENTITY == 'coupon':
-                for coupon in Util.coupons:
-                    to_write.append(coupon) 
+            if self.config.get('include_deleted') not in ['false','False', False]:
```

Comment:
Take this code in the nested scope of the above condition

Reply:
Done